caddyhttp/proxy/reverseproxy.go

```diff
@@ -252,8 +252,25 @@ func (rp *ReverseProxy) copyResponse(dst io.Writer, src io.Reader) {
 	io.CopyBuffer(dst, src, buf.([]byte))
 }
 
+var skipHeaders = map[string]struct{}{
```

Comment:
Could we add a comment referring to this comment (or the linked nginx docs page): https://github.com/mholt/caddy/pull/1112#discussion_r80092582 -- which explains why this map exists.